src/lightning_app/cli/cmd_install.py

```diff
@@ -14,7 +15,117 @@
 logger = Logger(__name__)
 
 
-def gallery_component(name: str, yes_arg: bool, version_arg: str, cwd: str = None) -> None:
+@click.group(name="install")
+def install() -> None:
+    """Install Lightning AI selfresources."""
+    pass
+
+
+@install.command("app")
+@click.argument("name", type=str)
+@click.option(
+    "--yes",
+    "-y",
+    is_flag=True,
+    help="disables prompt to ask permission to create env and run install cmds",
+)
+@click.option(
+    "--version",
+    "-v",
+    type=str,
+    help="Specify the version to install. By default it uses 'latest'",
+    default="latest",
+    show_default=True,
+)
+@click.option(
+    "--overwrite",
+    "-f",
+    is_flag=True,
+    default=False,
+    help="When set, overwrite the app directory without asking if it already exists.",
+)
+def install_app(name: str, yes: bool, version: str, overwrite: bool = False) -> None:
+    _install_app_command(name, yes, version, overwrite=overwrite)
+
+
+@install.command("component")
+@click.argument("name", type=str)
+@click.option(
+    "--yes",
+    "-y",
+    is_flag=True,
+    help="disables prompt to ask permission to create env and run install cmds",
+)
+@click.option(
+    "--version",
+    "-v",
+    type=str,
+    help="Specify the version to install. By default it uses 'latest'",
+    default="latest",
+    show_default=True,
+)
+def install_component(name: str, yes: bool, version: str) -> None:
+    _install_component_command(name, yes, version)
+
+
+def _install_app_command(name: str, yes: bool, version: str, overwrite: bool = False) -> None:
+    if "github.com" in name:
+        if version != "latest":
+            logger.warn(
+                f"The provided version {version} isn't the officially supported one. "
+                f"The provided version will be ignored."
```

Comment:
Small wording suggestion:
```suggestion
                "When installing from GitHub, only the 'latest' version is supported. "
                f"The provided version ({version}) will be ignored."
```